text/0000-app-image-extensions.md

```diff
@@ -140,27 +146,68 @@ Attempting to rebase an app directly after it's been upgraded is not permitted a
 
 ## Specification
 
-Platforms may extend app images using the above interface by running the `extend` executable as root in a new container and creating an image from the result.
-Extending an image should generate a single layer.
-This should happen prior to the normal CNB build or rebase process.
-The `*-metadata-toml-file` file is used to determine what packages to install.
-The `cache-dir` directory is used to cache package databases (e.g. to speed up `apt-get update`).
+Platforms may extend app images using the above interface by running the root buildpacks and creating an image from the result. This should happen prior to the normal CNB build or rebase process.
 
 ## UX
 
 `pack upgrade` (on base image)
-1. See if new base image available, if so proceed to (3). Otherwise, (2).
-2. Run status, if updates available, proceed to (3). Otherwise, stop.
-3. Run extend and publish new image. Store extension metadata on images. 
+1. See if new base image available, if so pull the new base image(s)
+1. Run the extension buildpack(s) against each base image.
 
 If running `pack upgrade` on a builder, perform operation against both builder and run images and publish both.
-If running `pack upgrade` on an app, perform operation against run image (with ephemeral output into app repo), then `pack rebase`.  
+If running `pack upgrade` on an app, perform operation against run image (with ephemeral output into app repo), then `pack rebase`.
+
+## Example: Imagemagick Buildpack
+
+For a buildpack that installs the Imagemagick package, the `buildpack.toml` would look like:
+
+```toml
+[buildpack]
+id = "example/imagemagick"
+privileged = true
+```
+
+The `bin/build` would look like:
+
+```bash
+#!/usr/bin/env bash
+
+apt update
+apt install imagemagick
+
+cat << EOF > ${1}/usr.toml
+paths = [ "/usr" ]
```

Comment:
The puts the onus on the buildpack developer to understand exactly what installing the imagemagick does to the file system. Also, it's not safe to install imagemagick on the build image and rebase it on to the run image (same problem as rebasing with updated versions of the same image).

Reply:
> The puts the onus on the buildpack developer to understand exactly what installing the imagemagick does to the file system.

I think this would be mitigated by the availability of community maintained `example/apt` and `example/yum` buildpacks. In some cases you might have an `example/imagemagick` buildpack if the package required some specific logic.

> Also, it's not safe to install imagemagick on the build image and rebase it on to the run image

I think running the buildpack on both the build and run images covers this right?